docs/design/eth1.md

```diff
@@ -0,0 +1,380 @@
+ex: https://geth.ethereum.org/docs/developers/dapp-developer/native-bindings
+
+
+### ETH1 API
+
+- Seperate event DB handling and memory actions.
+
+```golang
+//option 1 
+
+client := eth1client.New(grpcURL, contractADDR, contractABI, contractBlock)
+
+
+type EventBatch struct {
+    BlockNumber bigint
+    Events []Event
+}
+
+// option 1
+func EventsBatcher() <-chan EventBatch{} {
+batches := make(chan EventBatch{})
+    go func() {
+        blockNumber := begin
+
+        batch := EventBatch{blockNumber}
+
+        for log,err := client.StreamLogs(begin); err == nil         {
+            if log.BlockNumber > blockNumber {
+                batches <- batch
+                blockNumber = log.BlockNumber
+                batch = EventBatch{log.BlockNumber}
+            }
+
+            batch.Events = append(batch.Events, log)
+        }
+    }
+    return batches
+}
+
+// option2
+func EventsBatcher() <-chan EventBatch{} {
+batches := make(chan EventBatch{})
+    go func() {
+        blockNumber := begin
+
+        batch := EventBatch{blockNumber}
+
+        for block,err := client.WaitForFinalizedBlockAfter(begin); err == nil         {
+            if block.BlockNumber > blockNumber {
+                // extract relevant events from block
+                // add events to batch
+                batches <- batch
+            }
+
+        }
+    }
+    return batches
+}
+```
+
+### EventDataHandler needs DB Access
+
+```golang
+
+for {
+
+select {
+    case batch := <-batches:
+        StartDBTX()
+        tasks := handleBatch() // tasks are added to list of operations that should be done after db change
+        UpdateSyncOffset()
+        DoneDBTX() 
+        tasksChannel <- tasks// shot meaningfull event aftermath
+    case ctx.Done():
+        return 
+}
+
+
+
+```
+
+### BadgerDB Tx example
+
+```golang
+txn := badger.NewTransaction(true)
+
+for _, e := range entries {
+    if err := txn.Set(e.Key, e.Value); err != nil {
+    // XXX DB Error!, Rewind, retry, or quit
+    }
+}
+if err := txn.Commit(); err != nil {
+    // XXX DB Error!, Rewind, retry, or quit
+}
+```
+
+
+### EL sync
+
+if the EL node is not synced, we should stop event processing and signal the node to crash becasue we can't resume.
+a synecd EL node is a prerequisite to run SSV.
+
+After crash the SSV node will launch and will wait and retry until it synced, so it won't be in a crash loop.
+
+### Observability
+
+- Event processing time
+- malformed events
+- last processesd block
+
+### TBD integratio with ETH2
+
+---
+
+## Implementation details
+
+### Eth1Client
+
+Interacts with the underlying eth1 client and returns data prepared in a convenient format.
+
+#### Has methods:
+
+##### Exported:
+
+- `FetchPastLogs(ctx context.Context, fromBlock uint64) ([]ethtypes.Log, error)`
+
+Fetches past blocks.
+
+- `StreamLogs(ctx context.Context) <-chan ethtypes.Log`
```

Comment:
Instead of FilterLogs call custom func that calls FilterLogs multiple times and batches